### ISSUE TITLE
Disabling scroll restoration when doing `history.back()` after `pushState()`

### ISSUE THREAD

**Issue by aradalvand** [sveltejs/kit#11452]
### Describe the problem

I'm trying to use the new [shallow routing feature](https://kit.svelte.dev/docs/shallow-routing) to, for example, make it possible for users to close an app-like sidebar by using the back button, which I'd presume is the quintessential use case for this feature.

Things are working well apart from the fact that closing the sidebar (i.e. `history.back()`) often changes the scroll position in an annoying fashion (similar to how it's done upon page navigation), but which in this case very much feels disruptive and inappropriate UX-wise.

### Describe the proposed solution

I'm still not sure if it's the browser doing this on its own or if it's something SvelteKit is internally doing pr at least has control over; if it's the latter, then it would be useful if, for example, `pushState` received a option similar to `goto`'s `noscroll`, which disables this behavior.

### Importance

nice to have

**Comment by PatrickG** [comment]
This should be fixed by #11421

**Comment by aradalvand** [comment]
@PatrickG No, I don't want ANY scroll restoration. This is not the same as #11355.

**Comment by stefanVaporTrade** [comment]
I'm having the same issue when using replaceState.  I am capturing a product carousels scroll x position beforeNavigate and replacing the current state then on history.back() recalling that state and scrolling the carousel back to its last know position.

When using pushState for modals and drawers I am locking the body of the site by pushing a <style> body overflow:hidden to the head when the draw is open so I assume this is why I'm not getting the scroll issue elsewhere.

https://github.com/sveltejs/kit/assets/120146706/d8c88431-8871-4697-84e5-95cc93da08a6

**Comment by dummdidumm** [comment]
I can't reproduce this. If I scroll down a page, do a shallow navigation using pushState, then either type `history.back()` in the browser or use my mouse back button, in both cases it keeps the scroll position.

Please provide a reproduction

**Comment by Nic-Polumeyv** [comment]
On 2.0.7 (current when this was filed): scroll to 800, reload, scroll to 1500, `pushState`, `history.back()` jumps to 800. `pushState` didn't refresh the scroll snapshot for the entry you return to, so `back()` restored a stale one from sessionStorage. It needed a stale snapshot to exist, hence the intermittent reports. #11540 fixed it in 2.0.8, and the same probe on 2.70.0 and 3.0.0-next.9 stays at 1500.

If the page scrolls while the overlay is open (1500, push, scroll to 3000, back), current versions restore to 1500. Probe with all three pinned versions: https://github.com/Nic-Polumeyv/sveltekit-pushstate-scroll-probe
